src/modules/emote_autocomplete/hooks/ChatInput.jsx

```diff
@@ -0,0 +1,39 @@
+import {useEffect, useState} from 'react';
+import emoteMenuViewStore from '../../../common/stores/emote-menu-view-store.js';
+import {getAutocompletable} from '../../../utils/autocomplete.js';
+
+function handleChatInput() {
+  const focusedWord = getAutocompletable();
+
+  if (focusedWord != null) {
+    const strippedFocusedWord = focusedWord.replace(/\s|:/g, '');
+    const foundEmotes = emoteMenuViewStore.search(strippedFocusedWord);
+    return foundEmotes.map((emote) => emote.item);
+  }
+
+  return [];
+}
+
+export default function useChatInput(chatInputElement) {
+  const [emotes, setEmotes] = useState([]);
+
+  useEffect(() => {
+    function keydownCallback() {
+      const updateEmotes = () => setEmotes(handleChatInput());
+
+      if (!emoteMenuViewStore.isLoaded()) {
+        emoteMenuViewStore.once('updated', updateEmotes);
```

Comment:
every key down will subscribe to this, which adds a ton of additional listeners that will call updateEmotes when the load occurs. is there a way to detect this and load once only on hook mount? should the hook be listening itself?